### DISCUSSION TITLE
Tabular Editor define DataType

### DISCUSSION THREAD

**Discussion by TedCStack** [TabularEditor/TabularEditor3#473] [🙋♀️TE3 Q&A]
Is there a way to for a C+ script to define the DataType for a measure?

**Comment by otykier** [comment]
Hi @TedCStack 
Sorry for the late reply. You cannot directly set the DataType property on measures (it is read-only), since the DataType is inferred from the DAX expression on the measure. So if you need to convert a calculation to a certain type, you can use the DAX [CONVERT](https://dax.guide/convert) function. And in that case, you could write a C# script to modify the expression as follows:

```csharp
// Convert all selected measures to STRING, by wrapping their expression in a call to the DAX CONVERT-function:
foreach(var m in Selected.Measures)
{
    m.Expression = "CONVERT(" + m.Expression + ", string");
}
```
